File: 10/mschaffer.py
n = 2000000
source = range(2, n + 1)
primes = []
i = 0
max = n - 1

while i < max:
  prime = source[i]
  if prime:
    primes.append(prime)
    j = i
    while j < max:
      source[j] = False
      j += prime
  i += 1
  
print sum(primes)

File: 10/mschaffer.rb
n = 2_000_000
list = (2..n).to_a
primes = []
i = 0
max = n - 1

while i < max
  prime = list[i]
  if prime
    primes << prime
    j = i
    while j < max
      list[j] = nil
      j += prime
    end
  end
  i += 1
end

# more straightforward, but gc kills you
# while !list.empty?
#   prime = list.first
#   primes << prime
#   list = list.select { |n| n % prime != 0 }
# end

p primes.select{|n|n}.inject &:+
